cmd/pcr0tool/commands/dumpregisters/helpers/get_registers.go

```diff
@@ -0,0 +1,98 @@
+package helpers
+
+import (
+	"fmt"
+	"io/ioutil"
+	"runtime"
+
+	"github.com/9elements/converged-security-suite/v2/pkg/errors"
+	"github.com/9elements/converged-security-suite/v2/pkg/registers"
+	"github.com/9elements/go-linux-lowlevel-hw/pkg/hwapi"
+)
+
+type getRegistersConfig struct {
+	LocalhostByDefault bool
+	OverrideTXTPublic  string
+}
+
+// GetRegistersOption is an option for function GetRegisters
+type GetRegistersOption interface {
+	apply(*getRegistersConfig) error
+}
+
+// OptLocalhostByDefault an option to GetRegisters which signals to extract
+// data from localhost if no other sources are provided.
+type OptLocalhostByDefault bool
+
+func (opt OptLocalhostByDefault) apply(cfg *getRegistersConfig) error {
+	cfg.LocalhostByDefault = bool(opt)
+	return nil
+}
+
+// OptTXTPublic an option to GetRegisters to provide a path to a file
+// with TXT Public space dumped.
+type OptTXTPublic string
+
+func (opt OptTXTPublic) apply(cfg *getRegistersConfig) error {
+	cfg.OverrideTXTPublic = string(opt)
+	return nil
+}
+
+type getRegistersOptions []GetRegistersOption
+
+// Config converts a bunch of options to a config.
+func (opts getRegistersOptions) Config() (getRegistersConfig, error) {
+	var cfg getRegistersConfig
+	for _, opt := range opts {
+		if err := opt.apply(&cfg); err != nil {
+			return cfg, fmt.Errorf("unable to apply option %T:%v due to error: %w", opt, opt, err)
+		}
+	}
+	return cfg, nil
+}
+
+// GetRegisters extracts status registers.
+func GetRegisters(
+	options ...GetRegistersOption,
+) (registers.Registers, error) {
```

Comment:
Can this be in one line? I dont see a mass of arugments passed that could lead to confusion, but maybe you have your reasons and want to share? :)

Reply:
Answered in: https://github.com/9elements/converged-security-suite/pull/325#discussion_r854133968